src/base/settingsstorage.cpp

```diff
@@ -188,8 +185,8 @@ bool SettingsStorage::save()
 
 #ifndef Q_OS_MAC
     settings->sync(); // Important to get error status
-    QString newPath = settings->fileName();
     QSettings::Status status = settings->status();
+    QString newPath = settings->fileName();
```

Comment:
No, you get `QSettings::Status` after `sync()`, so these 2 *should* placed together.
Minor detail though...